shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/strategies/dataset/GenericDatasetStrategy.java

```diff
@@ -0,0 +1,118 @@
+package org.shanoir.ng.importer.strategies.dataset;
+
+import org.dcm4che3.data.Attributes;
+
+import org.shanoir.ng.dataset.modality.GenericDataset;
+import org.shanoir.ng.dataset.model.CardinalityOfRelatedSubjects;
+import org.shanoir.ng.dataset.model.DatasetExpression;
+import org.shanoir.ng.dataset.model.DatasetMetadata;
+import org.shanoir.ng.dataset.model.DatasetModalityType;
+import org.shanoir.ng.dataset.model.ProcessedDatasetType;
+import org.shanoir.ng.dicom.DicomProcessing;
+import org.shanoir.ng.importer.dto.Dataset;
+import org.shanoir.ng.importer.dto.DatasetsWrapper;
+import org.shanoir.ng.importer.dto.ExpressionFormat;
+import org.shanoir.ng.importer.dto.ImportJob;
+import org.shanoir.ng.importer.dto.Serie;
+import org.shanoir.ng.importer.strategies.datasetexpression.DatasetExpressionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GenericDatasetStrategy implements DatasetStrategy<GenericDataset> {
+	
+	@Autowired
+	DicomProcessing dicomProcessing;
+	
+	@Autowired
+	DatasetExpressionContext datasetExpressionContext;
+	
+	@Override
+	public DatasetsWrapper<GenericDataset> generateDatasetsForSerie(Attributes dicomAttributes, Serie serie,
+			ImportJob importJob) throws Exception {
+		DatasetsWrapper<GenericDataset> datasetWrapper = new DatasetsWrapper<>();
+		/**
+		 * retrieve number of dataset in current serie if Number of dataset > 1 then
+		 * each dataset will be named with an int at the end of the name. else the is
+		 * only one dataset => no need for extension.
+		 */
+		int datasetIndex;
+		if (serie.getDatasets().size() > 1) {
+			datasetIndex = 1;
+		} else {
+			datasetIndex = -1;
+		}
+
+		for (Dataset anyDataset : serie.getDatasets()) {
```

Comment:
Hi Jean-Come, I think it will be interesting to see in the import ms, how he "calculates" the datasets as they are not MR in this case. I have not yet verified it myself.

Reply:
I think the dicom structure is quite the same for all modalities, but I didn't check in import ms too.
I followed more or less what yao did for CT

Reply:
Hi Jean-Come, I checked: the method **addImageSeparateDatasetsInfo** today used for all imports is quite specific for MR(+Enhanced) only, we will have to verify with the dicom specs for each modality, what remains or not, as the code starts slightly to mix here (more easy for me as I originally migrated the import code into sh-ng and refactored a lot in comparison to sh-old). Same for this method: **checkIsMultiFrame**. It works yes, but is not really clean for multi-modality yet. Nothing urgent, but good to clean. Let's discuss in daily, with kind regards, Michael